pandas/tests/indexes/interval/test_interval_specific.py

```diff
@@ -0,0 +1,222 @@
+    """
+    # create scalar fixtures based on the class defined valid/invalid tuples
+    cls.scalars_valid = make_scalar_fixture(cls.scalar_tuples_valid)
+    cls.scalars_invalid = make_scalar_fixture(cls.scalar_tuples_invalid)
+
+    # create valid intervals fixture (only exact matches are valid)
+    interval_tuples_valid = [
+        (c, c, l, r) for c in 'lrbn' for l, r in set(cls.tuples)]
+    cls.intervals_valid = make_interval_fixture(interval_tuples_valid)
+
+    # create invalid intervals fixture (non-exact matches are invalid)
+    interval_tuples_invalid = generate_invalid_intervals(cls)
+    cls.intervals_invalid = make_interval_fixture(interval_tuples_invalid)
+    return cls
+
+
+class Base(object):
+    """
+    Common tests for IntervalIndex operations that differ depending on the
+    structure of the underlying intervals, e.g. monotonic vs. non-monotonic
+    """
+
+    @pytest.fixture(params=['left', 'right', 'both', 'neither'])
+    def closed(self, request):
+        """Valid values of 'closed' for an InteralIndex"""
+        return request.param
+
+    @pytest.fixture
+    def index(self, closed):
+        """
+        Create an IntervalIndex from the class specified tuples
+        """
+        return IntervalIndex.from_tuples(self.tuples, closed=closed)
+
+    def test_get_loc_scalar(self, scalars_valid):
+        index, scalar = scalars_valid
+        result = index.get_loc(scalar)
+        expected = [i for i, iv in enumerate(index) if scalar in iv]
+        if len(expected) == 1:
+            assert result == expected[0]
+        else:
+            expected = np.array(expected, dtype='int64')
+            tm.assert_numpy_array_equal(result, expected)
+
+    def test_get_loc_scalar_errors(self, scalars_invalid):
+        index, scalar = scalars_invalid
+        with pytest.raises(KeyError):
+            index.get_loc(scalar)
+
+    @pytest.mark.skip()
+    def test_get_loc_interval(self, intervals_valid):
+        index, interval = intervals_valid
+        result = index.get_loc(interval)
+        expected = [i for i, iv in enumerate(index) if interval == iv]
+        if len(expected) == 1:
+            assert result == expected[0]
+        else:
+            expected = np.array(expected, dtype='int64')
+            tm.assert_numpy_array_equal(result, expected)
+
+    @pytest.mark.skip()
+    def test_get_loc_interval_errors(self, intervals_invalid):
+        index, interval = intervals_invalid
+        with pytest.raises(KeyError):
+            index.get_loc(interval)
+
+
+class TestNonOverlappingMonotonic(Base):
+    """Tests specifics to non-overlapping monotonic IntervalIndex"""
+    # tuples from which to create the underlying IntervalIndex
+    tuples = [(0, 1), (2, 3)]
+
+    # valid scalar tuples to test, of the from (closed abbr, scalar)
+    # (a scalar is valid if it's contained in an interval in the index)
+    scalar_tuples_valid = [
+        ('l', 0), ('l', 0.5), ('l', 2), ('l', 2.5),
```

Comment:
Could remove some of these to eliminate redundancy, e.g. `('l', 0.5)` and `('l', 2.5)` are testing fundamentally the same thing.